lib/cocoapods-rome/post_install.rb

```diff
@@ -4,39 +4,75 @@
+
+    if build_xcframework
+      build_xcframework([device_lib, simulator_lib], build_dir, module_name)
+    else
+      executable_path = "#{build_dir}/#{root_name}"
+      build_universal_framework(device_lib, simulator_lib, build_dir, executable_path, module_name)
+    end
 
-    next unless File.file?(device_lib) && File.file?(simulator_lib)
+    FileUtils.rm device_lib if File.file?(device_lib)
+    FileUtils.rm simulator_lib if File.file?(simulator_lib)
+  end
+end
 
-    lipo_log = `lipo -create -output #{executable_path} #{device_lib} #{simulator_lib}`
-    puts lipo_log unless File.exist?(executable_path)
+def build_for_macos_platform(sandbox, build_dir, target, flags, configuration, build_xcframework=false)
+  target_label = target.cocoapods_target_label
+  xcodebuild(sandbox, target_label, flags, configuration)
 
-    FileUtils.mv executable_path, device_lib, :force => true
-    FileUtils.mv device_framework_lib, build_dir, :force => true
-    FileUtils.rm simulator_lib if File.file?(simulator_lib)
-    FileUtils.rm device_lib if File.file?(device_lib)
+  spec_names = target.specs.map { |spec| [spec.root.name, spec.root.module_name] }.uniq
+  spec_names.each do |root_name, module_name|
+    if build_xcframework
+      framework = "#{build_dir}/#{configuration}/#{root_name}/#{module_name}.framework"
+      build_xcframework([framework], build_dir, module_name)
+    end
   end
 end
 
-def xcodebuild(sandbox, target, sdk='macosx', deployment_target=nil, configuration)
+def xcodebuild(sandbox, target, sdk='macosx', deployment_target=nil, flags=nil, configuration)
   args = %W(-project #{sandbox.project_path.realdirpath} -scheme #{target} -configuration #{configuration} -sdk #{sdk})
+  args += flags unless flags.nil?  
   platform = PLATFORMS[sdk]
   args += Fourflusher::SimControl.new.destination(:oldest, platform, deployment_target) unless platform.nil?
   Pod::Executable.execute_command 'xcodebuild', args, true
 end
 
+def build_universal_framework(device_lib, simulator_lib, build_dir, destination, module_name)
+  device_executable = "#{device_lib}/#{module_name}"
+  simulator_executable = "#{simulator_lib}/#{module_name}"
+
+  raise Pod::Informative, 'Framework executables were not found in the expected location.' unless File.file?(device_executable) && File.file?(simulator_executable)
+
+  device_framework_lib = File.dirname(device_executable)
+  lipo_log = `lipo -create -output #{destination} #{device_executable} #{simulator_executable}`
+  puts lipo_log unless File.exist?(destination) 
+
+  FileUtils.mv destination, device_executable, :force => true
+  FileUtils.mv device_framework_lib, build_dir, :force => true
+end
+
+def build_xcframework(frameworks, destination, module_name)
+  args = %W(-create-xcframework -output #{destination}/#{module_name}.xcframework)
+
+  frameworks.each do |framework|
+    args += %W(-framework #{framework})
+  end
+
+  Pod::Executable.execute_command 'xcodebuild', args, true
+end
```

Comment:
I would suggest a couple of additions to this function. Sometimes we can have references to the same dependencies through other dependencies (e.g. we can have `RxSwift` but also `RxTest` which requires `RxSwift`) which can lead to duplicates and errors then.

```ruby
def build_xcframework(frameworks, build_dir, module_name)
  xcframework = "#{build_dir}/#{module_name}.xcframework"
  return if File.exist?(xcframework) 

  args = %W(-create-xcframework -output #{xcframework})

  frameworks.each do |framework|
    return unless File.exist?(framework) 
    args += %W(-framework #{framework})
  end

  Pod::Executable.execute_command 'xcodebuild', args, true
end
```

Reply:
cool. thanks. will include

Reply:
Hello guys!

Congrats to this pull request! Awesome work! 😍😍
I was testing this fork within my setup, and after adding the changes provided by Alex, it works perfectly given those changes also prevent some issues with aggregated targets, given some of them won't produce any frameworks and those checks will allow to detect that and skip the XCFramework generation.

I was wondering if you're planning to merge this soon, given that after going further on the setup within my project, I did flag some other major issues happening on the last steps (copy phase) from Rome. It would be great if I could create a pull request after those changes gets merged back in master. 🙏 🙏 🙏 

Again, great work! 🌟 🌟  This is a major step forward into actively support Rome.

Reply:
I've actually just found a pull request with such issue fixed here;

https://github.com/CocoaPods/Rome/pull/84/files#diff-1cf216328c39ad6fd0a36ecca66fb9870643fda2a42a9cd050bca52da8339edeR71

Reply:
Thanks for checking it out, @Ricowere! I'll add it as a [suggestion](https://github.com/CocoaPods/Rome/pull/90/files/2f6d329172af2e52db795f5bfc391418ad31139a#diff-1cf216328c39ad6fd0a36ecca66fb9870643fda2a42a9cd050bca52da8339ede). Also I don't think [we're aware of merging plans](https://github.com/CocoaPods/Rome/pull/90#issuecomment-729889476), I'd gladly know it too :)